### PR TITLE
Allow support for attachments

### PR DESCRIPTION
**Pull request by rajumsys**
#3

### CODE REVIEW

File: examples/transmission/with_attachment.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        name: 'Sparky.txt', 
+        type: 'text/plain', 
+        data: attachment 
+    }]
+}
+
+sp = SparkPost::Client.new() # pass api key or get api key from ENV
+
+# send message with attachment
+response = sp.transmission.send_message('rajuru@gmail.com', 'raju@test.spappteam.com', 'test email', '<h1>Message with attachment</h1>', options)
```
Comment (yepher) [inline examples/transmission/with_attachment.rb]: Are you sure you want to use statically defined email addresses here that seem to point to your personal account?

Comment (rajumsys) [inline examples/transmission/with_attachment.rb]: damn! i kept repeating this :(.

File: examples/transmission/with_attachment.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env ruby
+
+require 'base64'
+require_relative '../../lib/sparkpost'
+
+# read file and base64 encoding
+attachment = Base64.encode64(File.open(File.expand_path('../attachment.txt', __FILE__), 'r') { |f| f.read })
```
Comment (yepher) [inline examples/transmission/with_attachment.rb]: is `attachment.txt` really back one directory? It lives here `examples/transmission/attachment.txt` but maybe I am reading this wrong.

Comment (rajumsys) [inline examples/transmission/with_attachment.rb]: this is how `expand_path` works. 

http://apidock.com/ruby/File/expand_path/class#891-Require-file-from-the-same-folder

**Pull request merged**